ledger-core/v2/virtual/object/object.go

```diff
@@ -95,12 +111,45 @@ func (sm *SMObject) Init(ctx smachine.InitializationContext) smachine.StateUpdat
 	if !ctx.Publish(sm.Reference.String(), sdl) {
 		return ctx.Stop()
 	}
-	return ctx.Jump(sm.stepReadyToWork)
+
+	switch sm.initReason {
+	case InitReasonCTConstructor:
+		return ctx.Jump(sm.stepReadyToWork)
+	case InitReasonCTMethod:
+		return ctx.Jump(sm.stepGetObjectState) // TODO PLAT-294 - we will jump to stepWaitState
+	case InitReasonVStateReport:
+		return ctx.Jump(sm.stepWaitState)
+	default:
+		panic("Not implemented")
```

Comment:
this is wrong. you should use method from package `throw.NotImplemented`. However, in this particular case error should be `IlleagalValue`.